cpp/src/io/parquet/decode_fixed.cu

```diff
@@ -993,6 +1029,22 @@ CUDF_KERNEL void __launch_bounds__(decode_block_size_t, 8)
     return;
   }
 
+  // Exit early if the page does not need to be decoded
+  if (not page_mask[page_idx]) {
+    pp->num_nulls  = pp->num_rows;
+    pp->num_valids = 0;
```

Comment:
Separate from this PR, but I'm trying to remember how we ended up having both `num_nulls` and `num_valids`. One of them is redundant once we know `num_rows`

Reply:
Fwiw, I don't think we really use these `PageInfo` fields after decoding is complete. I only set these so they reflect the page being pruned.

Reply:
@etseidl